Framework/PythonInterface/mantid/plots/functions.py

```diff
@@ -0,0 +1,760 @@
+
+    For matrix workspaces with more than one spectra, either ``specNum`` or ``wkspIndex``
+    needs to be specified. Giving both will generate a :class:`RuntimeError`. There is no similar
+    keyword for MDHistoWorkspaces. These type of workspaces have to have exactly one non integrated
+    dimension
+    '''
+    if isinstance(workspace,mantid.dataobjects.MDHistoWorkspace):
+        (normalization,kwargs)=_getNormalization(workspace, **kwargs)
+        (x,y,dy)=_getMDData1D(workspace,normalization)
+        dx=None
+    else:
+        (wkspIndex, distribution, kwargs) = _getWkspIndexDistAndLabel(workspace, **kwargs)
+        (x, y, dy, dx) = _getSpectrum(workspace, wkspIndex, distribution, withDy=True, withDx=True)
+    _setLabels1D(axes, workspace)
+    return axes.errorbar(x, y, dy, dx, *args, **kwargs)
+
+
+def scatter(axes, workspace, *args, **kwargs):
+    '''
+    Unpack mantid workspace and render it with matplotlib. ``args`` and
+    ``kwargs`` are passed to :py:meth:`matplotlib.axes.Axes.scatter` after special
+    keyword arguments are removed. This will automatically label the
+    line according to the spectrum number unless specified otherwise.
+
+    :param axes:      :class:`matplotlib.axes.Axes` object that will do the plotting
+    :param workspace: :class:`mantid.api.MatrixWorkspace` or :class:`mantid.api.IMDHistoWorkspace`
+                      to extract the data from
+    :param specNum:   spectrum number to plot if MatrixWorkspace
+    :param wkspIndex: workspace index to plot if MatrixWorkspace
+    :param distribution: ``None`` (default) asks the workspace. ``False`` means
+                         divide by bin width. ``True`` means do not divide by bin width.
+                         Applies only when the the workspace is a MatrixWorkspace histogram.
+    :param normalization: ``None`` (default) ask the workspace. Applies to MDHisto workspaces. It can override
+                          the value from displayNormalizationHisto. It checks only if
+                          the normalization is mantid.api.MDNormalization.NumEventsNormalization
+
+    For matrix workspaces with more than one spectra, either ``specNum`` or ``wkspIndex``
+    needs to be specified. Giving both will generate a :class:`RuntimeError`. There is no similar
+    keyword for MDHistoWorkspaces. These type of workspaces have to have exactly one non integrated
+    dimension
+    '''
+    if isinstance(workspace,mantid.dataobjects.MDHistoWorkspace):
+        (normalization,kwargs)=_getNormalization(workspace, **kwargs)
+        (x,y,_)=_getMDData1D(workspace,normalization)
+    else:
+        (wkspIndex, distribution, kwargs) = _getWkspIndexDistAndLabel(workspace, **kwargs)
+        (x, y, _, _) = _getSpectrum(workspace, wkspIndex, distribution)
+    _setLabels1D(axes, workspace)
+    return axes.scatter(x, y, *args, **kwargs)
+
+
+def contour(axes, workspace, *args, **kwargs):
```

Comment:
Needs documentation of `AxisAligned` parameter in all of the `pcolor*` family. The paramer is not very pythonic either. Maybe `axisaligned` would be better.